ast/src/parsed/mod.rs

```diff
@@ -658,6 +671,22 @@ impl<R> Children<Expression<R>> for IfExpression<R> {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct LetStatementInsideBlock<Ref = NamespacedPolynomialReference> {
```

Comment:
What's the difference between this and a normal let statement?

Reply:
The other let statement is just an enum variant. The other difference is that here, we do not support types yet. Once add types, we can unify the two.

Reply:
Actually I could unify them now and just not parse the type scheme.

Reply:
ok, this gets a bit too complicated and I would like to defer that to later. The reason is that at parsing time, we have `Type<Expression<NamespacedPolynomialReference>>` and `Expression<NamespacedPolynomialReference>`, while after analysis we have `Type<u64>` and `Expression<Reference>`. Probably better to fix that by splitting the Expression type into multiple types.

Reply:
ok